mautic/mautic
# Form phone field validation as internationally-standardized format

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/1614
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

This bugfix/feature has been sponsored by @Webmecanik

[//]: # ( Required: )
#### Description:
Based on this feature request https://github.com/mautic/mautic/issues/1614 
This PR added validation based on https://github.com/giggsey/libphonenumber-for-php, already used in SMSBundle. 
Twillio require E.164 number formatting   https://support.twilio.com/hc/en-us/articles/223183008-Formatting-International-Phone-Numbers


#### Steps to test this PR:
1.  Create form
2.  Create Phone field. See properties tab and  option:  Validate as internationally-standardized format E.164. Enable it and also enable  Validation as  Required
![image](https://user-images.githubusercontent.com/462477/33431443-a3e31fd4-d5d4-11e7-9bf0-b74a827910f9.png)
3. Use form on your site.
4. See if form is submitted just after paste corect internation aformat (+421911111111 for example for Slovakia)
